generatedocs/compile/compile.go

```diff
@@ -69,10 +104,100 @@ func handleFlags() {
 		flag.PrintDefaults()
 		os.Exit(ExitCodeOk)
 	}
+
+	dataRoots = strings.Split(dataRootParam, ",")
+}
+
+func generateDocs(tmpl []byte, data *Data) []byte {
+	// Create template for generating documentation.go
+	documentationTemplate, err := template.New("documentationTemplate").Funcs(funcMap).Parse(string(tmpl))
+	if err != nil {
+		common.Exit(err, ExitCodeFailedParseTemplate)
+	}
+
+	buf := bytes.NewBuffer(nil)
+	if err := documentationTemplate.Execute(buf, data); err != nil {
+		common.Exit(err, ExitCodeFailedExecuteTemplate)
+	}
+
+	return buf.Bytes()
 }
 
-func compile() {
-	var data Data
+func mergeGroups(baseGroups []common.GroupsEntry, newGroups []common.GroupsEntry) []common.GroupsEntry {
+	// init new group lookup table
+	newGroupMap := make(map[string]common.GroupsEntry, len(newGroups))
+	for _, g := range newGroups {
+		newGroupMap[g.Name] = g
+	}
+
+	// init return value
+	mergedGroups := make([]common.GroupsEntry, 0, len(baseGroups)+len(newGroups))
+
+	// merge groups existing in base
+	for _, baseGroup := range baseGroups {
+		if newGroup, existInBase := newGroupMap[baseGroup.Name]; existInBase {
+			// mark new group as merged by deleting it from lookup table
+			delete(newGroupMap, baseGroup.Name)
+
+			// override string fields
+			if baseGroup.Description != "" {
```

Comment:
Why overwrite only when it's non-empty?

Reply:
No reason 🙈  it is supposed to be newGroup. Will make sure this is tested to, the tests are only with empty Descriptions etc.

Reply:
Alright :) Don't know how common of a case it'd be but I'm thinking one of these could be empty and the extending group would have it.

Reply:
Not so common I guess. But I think it makes more sense to override if extending group has it, which covers that case.